ir/src/main/scala/org/scalajs/core/ir/Types.scala

```diff
@@ -84,13 +84,14 @@ object Types {
   case object NullType extends Type
 
   /** Reference types (allowed for classOf[], is/asInstanceOf[]). */
-  sealed abstract class ReferenceType extends Type
+  sealed trait ReferenceType
```

Comment:
`AsInstanceOf(???, ClassType("scala.Int"))` (which would actually be `AsInstanceOf(???, ClassType("I"))`) is not valid IR. It would be `Unbox(???, 'I')` which has `tpe = IntType`.